File: GTK3/nim/helloworld/helloworld.nim
# requires the gintro library:
#   https://github.com/StefanSalewski/gintro

# run with:
#   nimble build && ./helloworld


import gintro/[gtk, gobject, gio]


proc buttonClicked (button: Button) =
  quit(QuitSuccess)


proc appActivate (app: Application) =
  let window = newApplicationWindow(app)
  window.title = "Hello World"
  window.defaultSize = (300, 300)
  let button = newButton("Hello World!")
  window.add(button)
  button.connect("clicked",  buttonClicked)
  window.showAll


proc main =
  let app = newApplication("org.gtk.HelloWorld")
  connect(app, "activate", appActivate)
  discard app.run


main()

File: GTK3/ocaml/helloworld/helloworld.ml
(*
 * requires the lablgtk library:
 *  https://github.com/garrigue/lablgtk
 *
 * run with:
 *  dune build && dune exec ./helloworld.exe
*)


let _ = GMain.init ()


let window = GWindow.window ~title:"Hello World" ~height:300 ~width:300 ()

let button = GButton.button ~label:"Hello World!" ~packing: window#add ()


let main () =
  window#connect#destroy ~callback:GMain.quit;
  button#connect#clicked ~callback:window#destroy;
  window#show ();
  GMain.main ()
;;

let _ = Printexc.print main ()
